### PR TITLE
Quick fix for Structured Dropout checkpointing

### PR DESCRIPTION
**Pull request by mingruimingrui**
Here's a quick fix for https://github.com/pytorch/fairseq/issues/1403.

To keep it short, this fix allows the user to checkpoint a translation model properly after applying layer pruning on a restored transformer file.

### CODE REVIEW

File: fairseq/models/fairseq_model.py [hunk 1]
```diff
@@ -68,6 +68,14 @@ def load_state_dict(self, state_dict, strict=True, args=None):
...
         self.upgrade_state_dict(state_dict)
         new_state_dict = prune_state_dict(state_dict, args)
+
+        # Since layers are now pruned, *_layers_to_keep are no longer needed.
+        # This is more of "It would make it work fix" rather than a proper fix.
+        if "encoder_layers_to_keep" in vars(args):
+            args.encoder_layers_to_keep = None
+        if "decoder_layers_to_keep" in vars(args):
+            args.decoder_layers_to_keep = None
+
```
Comment (myleott) [inline fairseq/models/fairseq_model.py]: Please move this to the end of prune_state_dict instead: https://github.com/pytorch/fairseq/blame/master/fairseq/checkpoint_utils.py#L397
